src/configurationProvider.ts

```diff
@@ -213,30 +215,94 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
             });
             return undefined;
         }
-        const pickItems = res.map((item) => {
-            let name = item.mainClass;
-            let details = `main class: ${item.mainClass}`;
-            if (item.projectName !== undefined) {
-                name += `<${item.projectName}>`;
-                details += ` | project name: ${item.projectName}`;
-            }
-            return {
-                description: details,
-                label: name,
-                item,
-            };
-        }).sort((a, b): number => {
-            return a.label > b.label ? 1 : -1;
-        });
+
+        const pickItems: IMainClassQuickPickItem[] = this.formatRecentlyUsedMainClassOptions(res);
+
         const selection = pickItems.length > 1 ?
             await vscode.window.showQuickPick(pickItems, { placeHolder: "Select main class<project name>" })
             : pickItems[0];
+
         if (selection && selection.item) {
+            this.debugHistory.updateMRUTimestamp(selection.item);
             return selection.item;
         } else {
             return undefined;
         }
     }
+
+    private isOpenedInActiveEditor(file: string): boolean {
+        const activeEditor: vscode.TextEditor = vscode.window.activeTextEditor;
+        const currentActiveFile: string = activeEditor ? activeEditor.document.uri.fsPath : undefined;
+
+        if (!file || !currentActiveFile) {
+            return false;
+        }
+
+        if (path.relative(file, currentActiveFile) === "") {
+            return true;
+        }
+
+        return false;
```

Comment:
can combine all the boolean operations like return file && currentActiveFile && path.relative(file, currentActiveFile) === "")

Reply:
fixed.